libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/BotFrameworkHttpAdapter.cs

```diff
@@ -62,6 +62,12 @@ public async Task ProcessAsync(HttpRequest httpRequest, HttpResponse httpRespons
             // deserialize the incoming Activity
             var activity = HttpHelper.ReadRequest(httpRequest);
 
+            if (string.IsNullOrEmpty(activity?.Type))
+            {
+                httpResponse.StatusCode = (int)HttpStatusCode.BadRequest;
+                return;
```

Comment:
Consider logging something. That way when support requests come in complaining "My Bot is returning Bad Request,and I'm not sure why" the truth is in the log files.